### ISSUE TITLE
Alias to source is broken if another alias is called

### ISSUE THREAD

**Issue by kubouch** [nushell/nushell#4838]
### Describe the bug

See repro. This one does not seem to break virtualenv tests but it potentially could in real usage.

### How to reproduce

Have two scripts:
```
# activate-foo.nu

alias foo = echo foo
alias deactivate-foo = (source deactivate-foo.nu)
```
```
# deactivate-foo.nu

hide foo
```

This sequence works:
```
> source ./activate-foo.nu
> deactivate-nu
> source ./activate-foo.nu
> deactivate-nu
...
```

This one doesn't:
```
> source ./activate-foo.nu
> foo
> deactivate-nu
Error: nu::parser::not_found (https://docs.rs/nu-protocol/0.59.1/nu-protocol/enum.ShellError.html#variant.NotFound)

  × Not found.
   ╭─[deactivate-foo.nu:1:1]
 1 │ hide foo
   ·      ─┬─
   ·       ╰── did not find anything under this name
   ╰────
```

### Expected behavior

It should work the same as the previous sequence

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.59.1                                   |
| branch             | main                                     |
| short_commit       | 14dc662e                                 |
| commit_hash        | 14dc662e505f4aa648240329250c125a6b4c859e |
| commit_date        | 2022-03-10 13:37:24 +00:00               |
| build_os           | linux-x86_64                             |
| rust_version       | rustc 1.59.0 (9d1b2106e 2022-02-23)      |
| rust_channel       | stable-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.59.0 (49d8809dc 2022-02-10)      |
| pkg_version        | 0.59.1                                   |
| build_time         | 2022-03-10 16:08:48 +02:00               |
| build_rust_channel | release                                  |
| features           | dataframe, default, trash, which, zip    |
| installed_plugins  | gstat, inc                               |

### Additional context

Related: https://github.com/nushell/nushell/issues/4822 https://github.com/nushell/nushell/issues/4839 https://github.com/nushell/nushell/pull/4835

**Comment by kubouch** [comment]
This issue changed: Now after sourcing `activate-foo.nu`, it's not possible to call `foo` anymore.

**Comment by kubouch** [comment]
Fixed by #4858

**Comment by kubouch** [comment]
Fixed by #4858
